Added two additional examples for the Media Service (#2046)
umbraco/UmbracoDocs: Reference/Management/Services/MediaService/Index.md

## Samples

### Creating a new folder

> To create a new folder at the root of the media archive, your code could look as:

Reviewer: ```suggestion
To create a new folder at the root of the media archive, your code could like the following:
```

```csharp
// Initialize a new media at the root of the media archive
IMedia folder = Services.MediaService.CreateMedia("My Folder", Constants.System.Root, Constants.Conventions.MediaTypes.Image);

// Save the folder
Services.MediaService.Save(folder);
```

For the `CreateMedia` method, the first parameter is the name of the folder to be created.

The second parameter is the ID of the parent media item. `Constants.System.Root` is a constant defined in Umbraco with the value of `-1`, which is used for indicating the root of the media archive. Instead of specifying the numeric ID of the parent, you may instead specify either a `Guid` ID or an `IMedia` instance representing the parent media.

> The third parameter is the alias of the media type. As Umbraco comes with a folder type by default, we can use the `Constants.Conventions.MediaTypes.Folder` constant to specify that the alias of the media type is `Folder`.

Reviewer: ```suggestion
The third parameter is the alias of the Media Type. As Umbraco comes with a Folder Type by default, we can use the `Constants.Conventions.MediaTypes.Folder` constant to specify that the alias of the Media Type is `Folder`.
```

In addition to the three mandatory parameters as shown above, you may also specify a numeric ID for a user to which the creation of the media should be attributed. If not specified, the media will be attributed to the user with ID `-1`.

### Creating a new media from an uploaded file

> The example below shows how to create a new file (on in this case, an image) from a HTTP upload. For illustrative purposes the example is a Razor view.

Reviewer: ```suggestion
The example below shows how to create a new file (in this case, an image) from a HTTP upload. For illustrative purposes the example is a Razor view.
```

Umbraco uses this instance to determine the type of the media you're creating, as well as handling a few things "under the hood" so you don't have to. For instance Umbraco will automatically set other properties relatd to the file - such as file size and image dimensions.
:::

> ### Creating a new media from a stream

Reviewer: Sounds a bit like you're missing an `item` here? 🤔 

```suggestion
### Creating a new media item from a stream
```

> Similar to specifying a `HttpPostedFileBase`as shown in the example above, you can instead specify a `Stream` for the contents of the file that should be created.

Reviewer: ```suggestion
Similar to specifying a `HttpPostedFileBase` as shown in the example above, you can instead specify a `Stream` for the contents of the file that should be created.
```

As an example, if you have a file on disk, you can open a new stream for a file on the disk, and then create a new media for that file in Umbraco:

```csharp
// Open a new stream to the file
using (Stream stream = File.OpenRead("C:/path/to/my-image.jpg"))
{

    // Initialize a new image at the root of the media archive
    IMedia media = Services.MediaService.CreateMedia("My image", Constants.System.Root, Constants.Conventions.MediaTypes.Image);

>     // Set the property value (Umbraco will handle the underlying magic)

Reviewer: Is it really magic though? 😅 
```suggestion
    // Set the property value (Umbraco will handle the underlying magic)
```

Author: I've now changed the text to "Umbraco will automatically update related properties" 😉

    media.SetValue(Services.ContentTypeBaseServices, Constants.Conventions.Media.File, "my-image.jpg", stream);

    // Save the media
    Services.MediaService.Save(media);

}
```

> Again Umbraco will make sure the necessary properties are updated

Reviewer: ```suggestion
Again Umbraco will make sure the necessary properties are updated.
```
